source-academy/modules
# [Sounds] Add audioplayer to SOUNDS library and made new library STEREOSOUNDS

# Add audioplayer to SOUNDS library and made new library STEREOSOUNDS

- Have transferred the SOUNDS library to modules,
- Have made new library called STEREOSOUNDS which produces stereo sounds. 
- Both Sound and StereoSound tabs, when spawned, contain an audio player that allows users to play their sound, pause their sound, adjust the volume of the sound and download the sound.

Fixes # (issue)

## Type of change

- [ ] Code quality imporvements

# How Has This Been Tested?

- Navigate to an instance of cadet-frontend, while running modules
- To test STEREOSOUNDS, run the following code:
```
import { make_stereo_sound, sine_sound, play, pan_mod } from "stereo_sounds";
const test = make_stereo_sound(t => math_sin(2 * 440 * math_PI * t), t => 0, 5);
const test2 = sine_sound(0.3, 70);
play(pan_mod(test2)(test));
```
- To test SOUNDS, run the following code: 
```
import { consecutively, triangle_sound, piano, violin, sine_sound, play, simultaneously } from "sounds";
const n1 = consecutively(list(triangle_sound(500,3), piano(66, 4)));
const n2 = consecutively(list(violin(60, 2), sine_sound(400, 5)));
play(simultaneously(list(n1, n2)));
```
- Both Sound tab and StereoSound tab should contain an audio player. Try playing, pausing, adjusting volume, downloading the sound as a file.
# Checklist:
